chia/cmds/db_upgrade_func.py

```diff
@@ -40,15 +46,64 @@ def db_upgrade_func(
         out_db_path = path_from_root(root_path, db_path_replaced)
         mkdir(out_db_path.parent)
 
-    convert_v1_to_v2(in_db_path, out_db_path)
-
-    if update_config:
-        print("updating config.yaml")
-        config = load_config(root_path, "config.yaml")
-        new_db_path = db_pattern.replace("_v1_", "_v2_")
-        config["full_node"]["database_path"] = new_db_path
-        print(f"database_path: {new_db_path}")
-        save_config(root_path, "config.yaml", config)
+    total, used, free = shutil.disk_usage(out_db_path.parent)
+    in_db_size = in_db_path.stat().st_size
+    if free < in_db_size:
+        no_free: bool = free < in_db_size * 0.6
+        strength: str
+        if no_free:
+            strength = "probably not enough"
+        else:
+            strength = "very little"
+        print(f"there is {strength} free space on the volume where the output database will be written:")
+        print(f"   {out_db_path}")
+        print(
+            f"free space: {free / 1024 / 1024 / 1024:0.2f} GiB expected about "
+            f"{in_db_size / 1024 / 1024 / 1024:0.2f} GiB"
+        )
+        if no_free and not force:
+            print("to override this check and convert anyway, pass --force")
+            return
+
+    try:
+        convert_v1_to_v2(in_db_path, out_db_path)
+
+        if update_config:
+            print("updating config.yaml")
+            config = load_config(root_path, "config.yaml")
+            new_db_path = db_pattern.replace("_v1_", "_v2_")
+            config["full_node"]["database_path"] = new_db_path
+            print(f"database_path: {new_db_path}")
+            save_config(root_path, "config.yaml", config)
+
+    except RuntimeError as e:
+        print(f"conversion failed with error: {e}.")
+    except Exception as e:
+
+        print(
+            textwrap.dedent(
+                f"""
+            conversion failed with error: {e}.
+            The target v2 database is left in place (possibly in an incomplete state)
+              {out_db_path}
+            If the failure was caused by a full disk, ensure the volumes of your
+            temporary- and target directory have sufficient free space."""
+            )
+        )
+        if platform.system() == "Windows":
+            temp_dir = None
+            # this is where GetTempPath() looks
+            # https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-gettemppatha
+            if "TMP" in os.environ:
+                temp_dir = os.environ["TMP"]
+            elif "TEMP" in os.environ:
+                temp_dir = os.environ["TEMP"]
+            elif "USERPROFILE" in os.environ:
+                temp_dir = os.environ["USERPROFILE"]
+            if temp_dir is not None:
+                print(f"your temporary directory may be {temp_dir}")
```

Comment:
I was expecting a similar message to UNIX where they are told how to specify the the temp directory to use.

Reply:
good point. presumably setting `TMP` is the simplest way